bindings/matrix-sdk-ffi/src/client_builder.rs

```diff
@@ -265,11 +508,16 @@ impl ClientBuilder {
         let builder = unwrap_or_clone_arc(self);
         let mut inner_builder = builder.inner;
 
-        if let (Some(base_path), Some(username)) = (builder.base_path, &builder.username) {
+        if let (Some(base_path), Some(username)) = (&builder.base_path, &builder.username) {
             // Determine store path
             let data_path = PathBuf::from(base_path).join(sanitize(username));
             fs::create_dir_all(&data_path)?;
 
+            inner_builder = inner_builder.sqlite_store(&data_path, builder.passphrase.as_deref());
+        } else if let Some(base_path) = &builder.base_path {
+            let data_path = PathBuf::from(base_path);
+            fs::create_dir_all(&data_path)?;
+
             inner_builder = inner_builder.sqlite_store(&data_path, builder.passphrase.as_deref());
         }
```

Comment:
This control flow could be simplified to something like:

```
if let Some(base_path) = ... {
  let data_path = if let Some(username) = ... { ... } else { ... };
 ...
}
```

Reply:
Ah indeed: https://github.com/matrix-org/matrix-rust-sdk/pull/3473/commits/18c8696883e3a76b63730a53175b565677f5f647.

Thanks.